src/ostree/ot-admin-builtin-status.c

```diff
@@ -200,10 +203,16 @@ ot_admin_builtin_status (int argc, char **argv, OstreeCommandInvocation *invocat
   if (booted_deployment)
     ostree_sysroot_query_deployments_for (sysroot, NULL, &pending_deployment, &rollback_deployment);
 
-  if (deployments->len == 0)
+  if (opt_query_booted)
     {
-      g_print ("No deployments.\n");
+      if (deployments->len == 0)
+        return FALSE;
```

Comment:
This must be
```suggestion
        return glnx_throw (error, "Not in a booted OSTree system");
```
or so.  We can't just `return FALSE` without setting an error.